components/tikv_util/src/thread_group.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+//! Provides util functions to manage share properties across threads.
+
+use std::cell::RefCell;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+
+#[derive(Default)]
+struct GroupPropertiesInner {
+    shutdown: AtomicBool,
+}
+
+#[derive(Default, Clone)]
+pub struct GroupProperties {
+    inner: Arc<GroupPropertiesInner>,
+}
+
+impl GroupProperties {
+    #[inline]
+    pub fn mark_shutdown(&self) {
+        self.inner.shutdown.store(true, Ordering::SeqCst);
+    }
+}
+
+thread_local! {
+    static PROPERTIES: RefCell<Option<GroupProperties>> = RefCell::new(None);
+}
```

Comment:
Could we use a global variable instead? So we don't need to pass it around threads.

Reply:
The problem of global variable is that it affects all threads, which can pollute tests as it will start multiple instance using threads.